src/App.tsx

```diff
@@ -1,26 +1,336 @@
+  }, [todos]);
+
+  const activeCount: number = useMemo(() => {
+    return todos.reduce((acc, todo) => {
+      if (todo.completed === false) {
+        return acc + 1;
+      }
+
+      return acc;
+    }, 0);
+  }, [todos]);
+
+  function trimTitle(text: string) {
+    return text.replace(/\s+/g, ' ').trim();
+  }
+
+  function showError(message: Error) {
+    if (errorTimeoutRef.current) {
+      clearTimeout(errorTimeoutRef.current);
+    }
+
+    setErrorMessage(message);
+    errorTimeoutRef.current = setTimeout(
+      () => setErrorMessage(Error.Default),
+      3000,
+    );
+  }
+
+  function filterToBool(filter: Filter) {
+    let boolFilter = null;
+
+    if (filter === Filter.Active) {
+      boolFilter = false;
+    } else if (filter === Filter.Completed) {
+      boolFilter = true;
+    }
+
+    return boolFilter;
+  }
+
+  function changeState(
+    id: number,
+    todosState: Dispatch<SetStateAction<Todo[]>>,
+    updatedTodo: Todo,
+  ) {
+    const filter = filterToBool(activeFilter);
+
+    todosState(prev => {
+      let changed = prev.map(todo => (todo.id === id ? updatedTodo : todo));
+
+      if (todosState === setFilteredTodos) {
+        changed = filterTodosByStatus(changed, filter);
+      }
+
+      return changed;
+    });
+  }
+
+  function handleTitleChange(newTitle: string, editingTodoId: number | null) {
+    const updateStatus = { title: newTitle };
+
+    return client
+      .patch<Todo>(`/todos/${editingTodoId}`, updateStatus)
+      .then(fetchedTodo => {
+        changeState(editingTodoId as number, setFilteredTodos, fetchedTodo);
+        changeState(editingTodoId as number, setTodos, fetchedTodo);
+      })
+      .catch(error => {
+        showError(Error.UpdateError);
+        throw error;
+      });
+  }
+
+  function handleTodoStatusChange(id: number, newStatus: boolean) {
+    setStatusChangeId(prev => [...prev, id]);
+    const updateStatus = { completed: newStatus };
+
+    return client
+      .patch<Todo>(`/todos/${id}`, updateStatus)
+      .then(fetchedTodo => {
+        changeState(id, setFilteredTodos, fetchedTodo);
+        changeState(id, setTodos, fetchedTodo);
+      })
+      .catch(() => showError(Error.UpdateError))
+      .finally(() => {
+        setStatusChangeId(prev => prev.filter(idParametr => idParametr !== id));
+      });
+  }
+
+  function setFocusOnAddInput() {
+    if (addTodoField.current !== null) {
+      addTodoField.current.focus();
+    }
+  }
+
+  useEffect(() => {
+    client
+      .get<Todo[]>(`/todos?userId=${USER_ID}`)
+      .then(fetchedTodos => {
+        setTodos(fetchedTodos);
+        setFilteredTodos(fetchedTodos);
+        setFocusOnAddInput();
+      })
+      .catch(() => showError(Error.LoadError));
+  }, []);
+
+  useEffect(() => {
+    if (tempTodo === null) {
+      /*tempTodo === null для того, не виконувати це два рази (бо стейт tempTodo спочатку змінюється на об'єкт а потім змінюється на null)*/
```

Comment:
Remove all comments. If you simplify your code and provide good names everyone understand what your code is doing without any comments